src/main/java/androidx/widget/TextView.kt

```diff
@@ -0,0 +1,156 @@
+        getUpdatedCompoundDrawable(bottom, compoundDrawables[3])
+    )
+}
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables.
+ *
+ * @see TextView.setCompoundDrawablesWithIntrinsicBounds
+ */
+fun TextView.updateCompoundDrawablesWithIntrinsicBounds(
+    left: Drawable? = drawableSentinel,
+    top: Drawable? = drawableSentinel,
+    right: Drawable? = drawableSentinel,
+    bottom: Drawable? = drawableSentinel
+) {
+    val drawables = updateDrawables(left, top, right, bottom, compoundDrawables)
+    setCompoundDrawablesWithIntrinsicBounds(drawables[0], drawables[1], drawables[2], drawables[3])
+}
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables.
+ *
+ * @see TextView.setCompoundDrawablesRelative
+ */
+@RequiresApi(17)
+fun TextView.updateCompoundDrawablesRelative(
+    start: Drawable? = drawableSentinel,
+    top: Drawable? = drawableSentinel,
+    end: Drawable? = drawableSentinel,
+    bottom: Drawable? = drawableSentinel
+) {
+    val drawables = updateDrawables(start, top, end, bottom, compoundDrawablesRelative)
+    setCompoundDrawablesRelative(drawables[0], drawables[1], drawables[2], drawables[3])
+}
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables. Use 0 if you do not want a Drawable there.
+ *
+ * @see TextView.setCompoundDrawablesRelativeWithIntrinsicBounds
+ */
+@RequiresApi(17)
+fun TextView.updateCompoundDrawablesRelativeWithIntrinsicBounds(
+    @DrawableRes start: Int = -1,
+    @DrawableRes top: Int = -1,
+    @DrawableRes end: Int = -1,
+    @DrawableRes bottom: Int = -1
+) {
+    val compoundDrawablesRelative = compoundDrawablesRelative
+    setCompoundDrawablesRelativeWithIntrinsicBounds(
+        getUpdatedCompoundDrawable(start, compoundDrawablesRelative[0]),
+        getUpdatedCompoundDrawable(top, compoundDrawablesRelative[1]),
+        getUpdatedCompoundDrawable(end, compoundDrawablesRelative[2]),
+        getUpdatedCompoundDrawable(bottom, compoundDrawablesRelative[3])
+    )
+}
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables.
+ *
+ * @see TextView.setCompoundDrawablesRelativeWithIntrinsicBounds
+ */
+@RequiresApi(17)
+fun TextView.updateCompoundDrawablesRelativeWithIntrinsicBounds(
+    start: Drawable? = drawableSentinel,
+    top: Drawable? = drawableSentinel,
+    end: Drawable? = drawableSentinel,
+    bottom: Drawable? = drawableSentinel
+) {
+    val drawables = updateDrawables(start, top, end, bottom, compoundDrawablesRelative)
+    setCompoundDrawablesRelativeWithIntrinsicBounds(
+        drawables[0], drawables[1], drawables[2], drawables[3]
+    )
+}
+
+private fun updateDrawables(
```

Comment:
I understand this is tempting to avoid code duplication. But doing isn't this involving creating a new `Array<Drawable?>` each time is it called? I would duplicate the code in each calling methods or find a way to avoid the array creation.

Reply:
It does create a new `Array<Drawable?>` per `updateXXXDrawablesXXX()` invocation but given this is a small, 4 index array, I'd personally think that this a good tradeoff for the benefit of losing code repetition. I am open to the suggestion to change it if the maintainers say so.

Reply:
The allocation doesn't matter much but the runtime matters more than making the code slightly better.